packages/broker/src/config/ConfigWizard.ts

```diff
@@ -1,254 +1,591 @@
-import { toEthereumAddress } from '@streamr/utils'
+import { ConfigFile, getDefaultFile } from './config'
 
-export interface PrivateKeyAnswers extends Answers {
-    generateOrImportPrivateKey: 'Import' | 'Generate'
-    importPrivateKey?: string
-}
+const MIN_BALANCE = utils.parseEther('0.1')
 
-export interface PluginAnswers extends Answers {
-    enabledApiPlugins: string[]
-    websocketPort?: string
-    mqttPort?: string
-    httpPort?: string
+export async function start(): Promise<void> {
+    log(`
+        >
+        > ***Welcome to the Streamr Network!***
+        > This Config Wizard will help you setup your node.
+        >
+        > The steps are documented here:
+        > *https://docs.streamr.network/guides/how-to-run-streamr-node#config-wizard*
+        >
+    `)
+
+    try {
+        const privateKey = await getPrivateKey()
+
+        const nodeAddress = new Wallet(privateKey).address
+
+        const network = await getNetwork()
+
+        const operator = await getOperatorAddress()
+
+        const operatorPlugins = operator ? {
+            operator: {
+                operatorContractAddress: operator,
+            },
+        } : {}
+
+        const { http, ...pubsubPlugins } = await getPubsubPlugins()
+
+        if (http) {
+            Object.assign(pubsubPlugins, {
+                http: {},
+            })
+        }
+
+        const storagePath = await getStoragePath()
+
+        const httpServer = http?.port ? { port: http.port } : undefined
+
+        const apiKey = Buffer.from(uuid().replace(/-/g, ''))
+            .toString('base64')
+            .replace(/[^\da-z]/gi, '')
+
+        let config: ConfigFile = {
+            $schema: formSchemaUrl(CURRENT_CONFIGURATION_VERSION),
+            client: {
+                auth: {
+                    privateKey,
+                },
+            },
+            plugins: {
+                ...operatorPlugins,
+                ...pubsubPlugins,
+            },
+            httpServer,
+            apiAuthentication: {
+                keys: [apiKey]
+            }
+        }
+
+        if (network === 'mumbai') {
+            config = getMumbaiConfig(config)
+        }
+
+        persistConfig(
+            storagePath,
+            config,
+        )
+
+        log(`
+            >
+            > ~ *Congratulations, you've setup your Streamr node!*
+            > Your node address is ${chalk.greenBright(nodeAddress)}
+            > Your node's generated name is ${chalk.greenBright(getNodeMnemonic(privateKey))}
+        `)
+
+        if (operator) {
+            const resume = progress((f) =>
+                style(`> Your node address has *${f} MATIC* _– checking balance…_`)
+            )
+
+            try {
+                const balance = await getNativeBalance(network, nodeAddress)
+
+                const content = `Your node address has *${utils
+                    .formatEther(balance)
+                    .replace(/\.(\d+)/, (f) => f.substring(0, 3))} MATIC*`
```

Comment:
Oh, this? Say you have a balance of `19.7465234786528937`. This makes it `19.74`.